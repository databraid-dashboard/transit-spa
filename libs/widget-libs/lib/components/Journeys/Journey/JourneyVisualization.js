var _jsxFileName = '../../src/components/Journeys/Journey/JourneyVisualization.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';

import { Icon, List, Step } from 'semantic-ui-react';

var JourneyVisualization = function JourneyVisualization(_ref) {
  var active = _ref.active,
      steps = _ref.steps;
  return React.createElement(
    List.Item,
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 7
      },
      __self: _this
    },
    React.createElement(
      Step.Group,
      {
        __source: {
          fileName: _jsxFileName,
          lineNumber: 8
        },
        __self: _this
      },
      steps.map(function (step, index) {
        var name = void 0;
        var title = void 0;

        if (step.mode === 'WALKING' && index !== 0) return null;

        if (step.mode.includes('WALKING')) {
          name = 'blind';
          title = 'Walk';
        }

        if (step.mode === 'TRANSIT') {
          name = step.instruction.includes('rail') ? 'subway' : 'bus';
          title = step.instruction.includes('rail') ? 'Train' : 'Bus';
        }

        return React.createElement(
          Step,
          { disabled: active, key: step.instruction, __source: {
              fileName: _jsxFileName,
              lineNumber: 26
            },
            __self: _this
          },
          React.createElement(Icon, { name: name, size: 'big', __source: {
              fileName: _jsxFileName,
              lineNumber: 27
            },
            __self: _this
          }),
          React.createElement(Step.Content, { title: title, __source: {
              fileName: _jsxFileName,
              lineNumber: 28
            },
            __self: _this
          })
        );
      })
    )
  );
};

JourneyVisualization.propTypes = {
  active: PropTypes.bool.isRequired,
  steps: PropTypes.arrayOf(PropTypes.object).isRequired
};

JourneyVisualization.defaultProps = {
  active: null,
  steps: [{}]
};

export default JourneyVisualization;