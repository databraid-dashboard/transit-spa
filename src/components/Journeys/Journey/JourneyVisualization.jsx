import React from 'react';
import PropTypes from 'prop-types';

import { Icon, List, Step } from 'semantic-ui-react';

import './journey-visualization.css';

function formatAgency(step) {
  let output = '';
  if (step.mode.includes('WALKING')) {
    output = 'Walk';
  }
  if (step.agency) {
    output = step.agency.replace(/Bay Area Rapid Transit/, 'BART');
  }
  return output;
}

function getInstructions(step) {
  let instr = '';
  if (!step) {
    return instr;
  }
  if (step.shortName) {
    if (step.instruction.startsWith('Bus ')) {
      instr = step.instruction.replace(/^Bus /, `Bus ${step.shortName} `);
    } else {
      instr = `${step.shortName} ${step.instruction}`;
    }
  } else {
    instr = step.instruction;
  }
  instr = instr.replace(/^Walk /, '');
  instr = instr.replace(/San Francisco International Airport/, 'SF Intl. Airport');
  return instr;
}

const JourneyVisualization = ({ active, steps }) =>
  (<List.Item>
    <Step.Group>
      {steps.map((step, index) => {
        let name;

        if (step.mode === 'WALKING' && index !== 0) return null;

        if (step.mode.includes('WALKING')) {
          name = 'blind';
        }

        if (step.mode === 'TRANSIT') {
          name = step.instruction.includes('rail') ? 'subway' : 'bus';
        }

        const uniqueKey = step.shortName + step.instruction;
        const agency = formatAgency(step);
        const instructions = getInstructions(step);

        return (
          <Step disabled={active} key={uniqueKey}>
            <Icon name={name} size="big" />
            <Step.Content title={agency} description={instructions} />
          </Step>
        );
      })}
    </Step.Group>
  </List.Item>);

JourneyVisualization.propTypes = {
  active: PropTypes.bool.isRequired,
  steps: PropTypes.arrayOf(PropTypes.object).isRequired,
};

JourneyVisualization.defaultProps = {
  active: null,
  steps: [{}],
};

export default JourneyVisualization;
