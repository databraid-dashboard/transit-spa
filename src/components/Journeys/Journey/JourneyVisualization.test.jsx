import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import JourneyVisualization from './JourneyVisualization';

describe('JourneyVisualization', () => {
  it('renders when given one step', () => {
    const component = shallow(
      <JourneyVisualization
        steps={[{ duration: '4 mins', instruction: 'Walk to Some St. Station', mode: 'WALKING' }]}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders when given multiple steps', () => {
    const component = shallow(
      <JourneyVisualization
        steps={[
          { duration: '4 mins', instruction: 'Walk to Some St. Station', mode: 'WALKING' },
          {
            agency: 'SFMTA',
            duration: '6 mins',
            headsign: 'Daly City',
            icon: undefined,
            instruction: 'Bus towards Daly City',
            line: '14',
            localIcon: undefined,
            longName: 'Mission',
            mode: 'TRANSIT',
            shortName: '14',
            vehicle: 'Bus',
          },
        ]}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
