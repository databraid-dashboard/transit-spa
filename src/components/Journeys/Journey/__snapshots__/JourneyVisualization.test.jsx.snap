// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JourneyVisualization renders when given multiple steps 1`] = `
<StepGroup
  className="journeyVizColumn"
>
  <Step>
    <TransitNode
      step={
        Object {
          "duration": "4 mins",
          "instruction": "Walk to Some St. Station",
          "mode": "WALKING",
        }
      }
    />
  </Step>
  <Step>
    <TransitNode
      step={
        Object {
          "agency": "SFMTA",
          "duration": "6 mins",
          "headsign": "Daly City",
          "icon": undefined,
          "instruction": "Bus towards Daly City",
          "line": "14",
          "localIcon": undefined,
          "longName": "Mission",
          "mode": "TRANSIT",
          "shortName": "14",
          "vehicle": "Bus",
        }
      }
    />
  </Step>
</StepGroup>
`;

exports[`JourneyVisualization renders when given one step 1`] = `
<StepGroup
  className="journeyVizColumn"
>
  <Step>
    <TransitNode
      step={
        Object {
          "duration": "4 mins",
          "instruction": "Walk to Some St. Station",
          "mode": "WALKING",
        }
      }
    />
  </Step>
</StepGroup>
`;
